internal/app/machined/pkg/runtime/v1alpha1/bootloader/grub/probe.go

```diff
@@ -0,0 +1,61 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package grub provides the interface to the GRUB bootloader: config management, installation, etc.
+package grub
+
+import (
+	"os"
+
+	"github.com/siderolabs/go-blockdevice/blockdevice/probe"
+
+	"github.com/siderolabs/talos/internal/pkg/mount"
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+)
+
+// Probe probes a block device for GRUB bootloader.
+func Probe() (*Config, error) {
+	// attempt to probe BOOT partition directly
+	dev, err := probe.GetDevWithPartitionName(constants.BootPartitionLabel)
+	if os.IsNotExist(err) {
+		// no BOOT partition, nothing to do
+		return nil, nil
```

Comment:
Maybe instead of `nil, nil` we should return an error which we can check at `Probe` callsite? Or at least document that this function can return both `nils` if no BOOT partition exists?

Reply:
I'm currently again re-structuring this differently after talking with Andrey